src/wallet/keystore.ts

```diff
@@ -0,0 +1,58 @@
+import {makeSurePathExists, resolveHome} from '../core/utils';
+import path from 'path';
+import config from 'config';
+import Wallet, {hdkey} from 'ethereumjs-wallet';
+import bip39 from 'bip39';
+
+const keystorePath = path.join(resolveHome(config.get('datadir')), config.get('wallet.keystore_path'));
+
+function getWalletFactory() {
+    let wallet: Wallet | undefined;
+    return (): Wallet => {
+        if (!wallet) {
+            const keystorePath = path.join(resolveHome(config.get('datadir')), config.get('wallet.keystore_path'));
+            makeSurePathExists(keystorePath);
+            const secretPhrase = require(path.join(keystorePath, 'key.json')).phrase;
+            const hdwallet = hdkey.fromMasterSeed(bip39.mnemonicToSeedSync(secretPhrase));
+            wallet = hdwallet.getWallet();
+        }
+        return wallet;
+    };
+}
+
+function getArweaveKeyFactory() {
+    let arweaveKey: string|undefined|null = null;
+    return (): string|undefined => {
+        if (arweaveKey === null) {
+            try {
+                arweaveKey = require(path.join(keystorePath, 'arweave.json'));
+            }catch(e){
+                arweaveKey = undefined;
+            }
+        }
+        return arweaveKey as string|undefined;
+    };
+}
+
+const getWallet = getWalletFactory();
+const getArweaveKey = getArweaveKeyFactory();
+
+export default {
+
+    getNetworkAddress() {
+        return `0x${getWallet().getAddress().toString('hex')}`;
+    },
+
+    getNetworkPublicKey() {
+        getWallet().getPublicKey().toString('hex');
+    },
+
+    getNetworkPrivateKey() {
+        getWallet().getPrivateKey().toString('hex');
+    },
+
+    getStorageKey() {
```

Comment:
Let's rename this method to `getStroragePrivateKey`. Also the `getStorageAddress` method is missing here. You can create one using `jwkToAddress` method available in the `arweave` package.